### PR TITLE
Release v1.24.0

### PR DESCRIPTION
**Pull request by AndrewSirenko**
**Is this a bug fix or adding new feature?**
Release

**What is this PR about? / Why do we need it?**
Release v1.24.0 upstream

**What testing is done?** 
CI 

Checklist:
    * Update changelog.md with content from stage 0 of runbook
    * readme.md latest version and install table
    * Makefile VERSION 
    * install.md all occurences of old version
    * charts/aws-ebs-csi-driver/Chart.yaml with new version & appVersion 
    * charts/aws-ebs-csi-driver/CHANGELOG.md with Helm changelog from stage 0
    * Update overlays by running make generate-kustomize

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+# v1.24.0
+### Notable Changes
+* Support clustered allocation with ext4 filesystems. This allows developers to enable [torn write prevention](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/storage-twp.html) on their dynamically provisioned volumes to improve the performance of I/O-intensive relational database workloads. ([#1706](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/1706), [@AndrewSirenko](https://github.com/AndrewSirenko))
+* Add volume limits for m7a, c7i, r7a, r7iz instance families ([#1742](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/1742), [@torredil](https://github.com/torredil))
+* Add volume attachment limits for c7a instance family ([#1776](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/1776), [@torredil](https://github.com/torredil))
```
Comment (torredil) [inline CHANGELOG.md:5]: np: Can we combine these into a single entry? ie:

Add volume limits for m7a, c7i, c7a, r7a, r7iz, instance families (#1742, #1776)

Comment (AndrewSirenko) [reply CHANGELOG.md:5]: Sure, I'll link to both PRs like so: ([#1742],[#1776], [@torredil])

**Pull request merged**